into/backends/ssh.py

```diff
@@ -219,7 +219,20 @@ def file_to_temp_ssh_file(typ, data, **kwargs):
 @convert.register(Temp(JSONLines), (Temp(SSH(JSONLines)), SSH(JSONLines)))
 @convert.register(Temp(JSON), (Temp(SSH(JSON)), SSH(JSON)))
 @convert.register(Temp(CSV), (Temp(SSH(CSV)), SSH(CSV)))
-def ssh_csv_to_temp_csv(data, **kwargs):
+def ssh_file_to_temp_file(data, **kwargs):
     fn = '.%s' % uuid.uuid1()
     target = Temp(data.subtype)(fn, **kwargs)
     return append(target, data, **kwargs)
+
+
+@convert.register(Temp(SSH(TextFile)), (TextFile, Temp(TextFile)))
+@convert.register(Temp(SSH(JSONLines)), (JSONLines, Temp(JSONLines)))
+@convert.register(Temp(SSH(JSON)), (JSON, Temp(JSON)))
+@convert.register(Temp(SSH(CSV)), (CSV, Temp(CSV)))
+def file_to_temp_ssh_file(data, **kwargs):
+    fn = '%s' % uuid.uuid1()
+    if isinstance(data, _Temp):
```

Comment:
i've been using `getattr(data, 'persistent_type', type(data))` for this. just a comment really, no action needed